### PR TITLE
Subtract fee from amount (rebase)

### PR DESCRIPTION
**Pull request by laanwj**
Rebase #4331
- Resolve conflicts w/ moved src/core.h
- Resolve conflict in CoinControlDialog::updateLabels function
- Port the RPC test to python

### CODE REVIEW

File: src/qt/coincontroldialog.cpp [hunk 1]
```diff
@@ -544,6 +545,11 @@ void CoinControlDialog::updateLabels(WalletModel *model, QDialog* dialog)
         // Fee
         nPayFee = CWallet::GetMinimumFee(nBytes, nTxConfirmTarget, mempool);
 
+        // in the subtract fee from amount case, we can tell if zero change already and subtract the bytes, so that fee calculation afterwards is accurate
```
Comment (cozz) [inline src/qt/coincontroldialog.cpp]: this code-block needs to be moved one line above as the variable nBytes is used there

**Comment by cozz** [comment]
nit: the second passage of the commit-message is now wrong, this was only true for the initial version of this pull-request, I'd say remove it, to avoid someone gets confused by it.

File: src/qt/coincontroldialog.cpp (hunk 1 above)
Comment (laanwj) [inline src/qt/coincontroldialog.cpp]: Thanks, good catch

**Comment by jonasschnelli** [comment]
needs rebase.

GUI and RPC looks good. Nice @cozz! Thanks for the rebase and rework @laanwj.

The new parameter for `sendmany` (json object) looks a bit as a overdose to me. A single boolean would maybe be sufficient. 

Tested ACK.

**Comment by laanwj** [comment]
With regard to future extensibility, I like the idea of adding a JSON structure to sendmany. Should have been done that in the first place instead of the long-winded positional command lines.

File: src/rpcwallet.cpp [hunk 2]
```diff
@@ -793,9 +804,9 @@ Value movecmd(const Array& params, bool fHelp)
 
 Value sendfrom(const Array& params, bool fHelp)
 {
-    if (fHelp || params.size() < 3 || params.size() > 6)
+    if (fHelp || params.size() < 3 || params.size() > 7)
         throw runtime_error(
-            "sendfrom \"fromaccount\" \"tobitcoinaddress\" amount ( minconf \"comment\" \"comment-to\" )\n"
+            "sendfrom \"fromaccount\" \"tobitcoinaddress\" amount ( minconf \"comment\" \"comment-to\" subtractfeefromamount )\n"
```
Comment (luke-jr) [inline src/rpcwallet.cpp]: Do we really want to extend deprecated methods?

**Comment by jonasschnelli** [comment]
Agreed. JSON structures as parameter (or as optional parameter) would be more convenient and extendable. `sandmany` and `sendfrom` are also marked for deprecation so i assume main focus should be GUI and `sendtoaddress`.

**Comment by laanwj** [comment]
eh, who is deprecating sendmany? If we were to keep one send* method, that would be the one.

**Comment by luke-jr** [comment]
@laanwj The JSON structure in the current PR only allows for a boolean for each recipient. Are you suggesting it be changed to a parameter object (eg, like getblocktemplate)? If so, I agree that seems like a better approach.

As far as multiple recipients sharing the fee, it seems more logical to be using an Array than an Object with only values of "true".

**Comment by jonasschnelli** [comment]
Re-ACK (including @luke-jr commits)

**Comment by cozz** [comment]
help-description of sendmany is now wrong after switching from object to array

**Comment by luke-jr** [comment]
@cozz What's wrong about it? Looks fine to me?

**Comment by cozz** [comment]
- "A json object with addresses and booleans" - its not an object but an array and it doesnt take booleans
- "Default for each address is false" - this sentence should be removed

**Comment by laanwj** [comment]
@luke-jr @cozz Added a commit that fixes the help message.

**Comment by laanwj** [comment]
I've squashed the two squashme commits, as well as @luke-jr's commit that reverts the changes to `sendfrom` (so that it' API not touched at all).

**Comment by jonasschnelli** [comment]
Retested ACK.
Binaries: https://builds.jonasschnelli.ch/pulls/5831/

bildschirmfoto 2015-03-13 um 17 35 32
bildschirmfoto 2015-03-13 um 17 35 07

**Comment by laanwj** [comment]
Thanks for testing @jonasschnelli

**Pull request merged**